apps/src/code-studio/pd/workshop_dashboard/workshops/overview/sections/WorkshopStatusSection.tsx

```diff
@@ -0,0 +1,190 @@
+import {Button, buttonColors} from '@code-dot-org/component-library/button';
+import Tags from '@code-dot-org/component-library/tags';
+import {
+  Heading2,
+  BodyFourText,
+  StrongText,
+  BodyThreeText,
+} from '@code-dot-org/component-library/typography';
+import {
+  Card,
+  CardContent,
+  CardHeader,
+  Box,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+} from '@mui/material';
+import classNames from 'classnames';
+import React, {useState} from 'react';
+
+import {WorkshopData} from '../../types';
+
+import styles from '../../workshop.module.scss';
+
+interface WorkshopStatusSectionProps {
+  workshop: WorkshopData;
+  onWorkshopUpdate: () => void;
+}
+
+export const WorkshopStatusSection: React.FC<WorkshopStatusSectionProps> = ({
+  workshop,
+  onWorkshopUpdate,
+}) => {
+  const [showEndDialog, setShowEndDialog] = useState(false);
+  const [isUpdating, setIsUpdating] = useState(false);
+
+  const handleStartWorkshop = async () => {
+    setIsUpdating(true);
+    try {
+      const response = await fetch(
+        `/api/v1/pd/workshops/${workshop.id}/start`,
+        {
+          method: 'PATCH',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        }
+      );
+
+      if (response.ok) {
+        onWorkshopUpdate();
+      } else {
+        console.error('Failed to start workshop');
+      }
+    } catch (error) {
+      console.error('Error starting workshop:', error);
+    } finally {
+      setIsUpdating(false);
+    }
+  };
+
+  const handleEndWorkshop = async () => {
+    setIsUpdating(true);
+    try {
+      const response = await fetch(`/api/v1/pd/workshops/${workshop.id}/end`, {
+        method: 'PATCH',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+
+      if (response.ok) {
+        onWorkshopUpdate();
+        setShowEndDialog(false);
+      } else {
+        console.error('Failed to end workshop');
+      }
+    } catch (error) {
+      console.error('Error ending workshop:', error);
+    } finally {
+      setIsUpdating(false);
+    }
+  };
```

Comment:
nit: I'm not super familiar with api requests in this part of the project, so more for context - do we have any middlewares for api requests? at least to reduce boilerplate code amount